guardrails/guard.py

```diff
@@ -650,6 +681,21 @@ def __parse(
             final_num_reasks = (
                 num_reasks if num_reasks is not None else 0 if llm_api is None else None
             )
+
+            # TODO: Add tracing: guard usage and user usage for Validator Hub
+            # Start a span for this guard parse
+            with self._hub_tracer.start_as_current_span("/guard_parse") as span:
+                # Inject the current context
+                self._hub_telemetry.inject_current_context()
+
+                span.set_attribute("guard_id", self._guard_id)
+                span.set_attribute("user_id", self._user_id)
+                span.set_attribute("llm_api", llm_api.__name__ if llm_api else "None")
+                span.set_attribute("custom_reask_prompt", self.reask_prompt is not None)
+                span.set_attribute(
+                    "custom_reask_instructions", self.reask_instructions is not None
+                )
```

Comment:
Same with this one, if we can refactor it into a util method that would be better

Reply:
Same as above